### PR TITLE
feat: enable vertical scrolling on demand

### PR DESCRIPTION
**Pull request by timowolf**
**What kind of change does this PR introduce?** (check one with "x")

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** The component doesn't support dynamic vertical scrolling.

**What is the new behavior?** There is an opt-in mode to support dynamic vertical scrolling.

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: projects/ngx-datatable/src/lib/components/body/body.component.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ import {
   EventEmitter,
   HostBinding,
   Input,
+  OnChanges,
```
Comment (fh1ch) [inline projects/ngx-datatable/src/lib/components/body/body.component.ts:8]: Is there a reason for this import here? It doesn't seem to be used?!

File: projects/ngx-datatable/src/lib/components/datatable.component.ts [hunk 2]
```diff
@@ -846,8 +854,22 @@ export class DatatableComponent implements OnInit, DoCheck, AfterViewInit, After
...
+    if (this.scrollbarV && !this.scrollbarVDynamic) {
       width = width - this.scrollbarHelper.width;
+
+    } else if (this.scrollbarVDynamic){
+      const scrollerHeight = this.bodyComponent?.scroller?.element.offsetHeight;
+      if (scrollerHeight && this.bodyHeight < scrollerHeight) {
+        width = width - this.scrollbarHelper.width;
+      }
+
+      if (this.headerComponent && this.headerComponent.innerWidth !== width){
```
Comment (fh1ch) [inline projects/ngx-datatable/src/lib/components/datatable.component.ts:866]: You've already used *optional chaining* above, so let's also do this here:

```suggestion
      if (this.headerComponent?.innerWidth !== width){
```

File: projects/ngx-datatable/src/lib/components/datatable.component.ts [hunk 3]
```diff
@@ -846,8 +854,22 @@ export class DatatableComponent implements OnInit, DoCheck, AfterViewInit, After
...
+        this.headerComponent.innerWidth = width;
+      }
+      if (this.bodyComponent && this.bodyComponent.innerWidth !== width){
```
Comment (fh1ch) [inline projects/ngx-datatable/src/lib/components/datatable.component.ts:869]: Same here:

```suggestion
      if (this.bodyComponent?.innerWidth !== width){
```

File: projects/ngx-datatable/src/lib/themes/material.scss [hunk 4]
```diff
@@ -51,6 +51,7 @@ $datatable-body-cell-color: rgba(0, 0, 0, 0.87) !default;
 $datatable-group-header-background: #f5f5f5 !default;
 $datatable-group-header-border-top-color: #d9d8d9 !default;
 $datatable-group-header-border-bottom-color: #d9d8d9 !default;
+$datatable-body-row-border-bottom-color: #d9d8d9 !default;
```
Comment (fh1ch) [inline projects/ngx-datatable/src/lib/themes/material.scss:54]: Just for my understanding: why is this new variable required? Can't it be derived from `$datatable-group-header-border-top-color`/`$datatable-group-header-border-bottom-color`?

**Review by fh1ch — CHANGES_REQUESTED** [review]
@timowolf really nice work, thanks a lot for the MR. Just some minor findings, otherwise good to go 💯

**Comment by fh1ch** [comment]
@timowolf I've also filled-out the MR template at the top (was empty). Please check if all of that is correct. Thanks!

**Comment by timowolf** [comment]
@fh1ch Many thanks for the review. We should setup strict listing rules and let the build fail on any issues.

File: projects/ngx-datatable/src/lib/themes/material.scss (hunk 4 above)
Comment (timowolf) [reply projects/ngx-datatable/src/lib/themes/material.scss:54]: We introduced the color to be able to see the border in the demos. Just with a white bg, we cannot easily see the sizing of the rows. A new variable is just to provide the option to color it differently to the group borders, if needed - wanted.

**Comment by timowolf** [comment]
@fh1ch Many thanks, review comments added and committed. Back to you.

**Review by fh1ch — APPROVED** [review]
@timowolf bit thanks for the rework, nothing else to add 🙇 

LGTM 👍

**Pull request merged**